src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -111,41 +111,52 @@ if (__DEV__) {
   } = require('ReactDebugFiberPerf');
 
   var didWarnAboutStateTransition = false;
+  var didWarnSetStateChildContext = false;
+  var ownerHasNoopWarning = {};
 
   var warnAboutUpdateOnUnmounted = function(
     instance: React$ComponentType<any>,
```

Comment:
Let's change this to receive `fiber` instead. Then we can use `getComponentName(fiber)` instead of writing the code to get it manually.

Reply:
When I changed it to fiber and tried, the tests failed because getComponentName(fiber) was returning null and not the component name itself for some reason. So had to revert it.